File: ios/Sources/CapacitorCalendarPlugin/Models/Inputs/DeleteCalendarInput.swift
import Capacitor

struct DeleteCalendarInput {
    private let id: String

    init(call: CAPPluginCall) throws {
        guard let id = call.getString("id") else {
            throw PluginError.idMissing
        }
        self.id = id
    }

    func getId() -> String {
        return id
    }
}

File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/DeleteCalendarInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.PluginError

data class DeleteCalendarInput(
    private val call: PluginCall,
) {
    val id: Long =
        call.getString("id")?.let { idString ->
            idString.toLongOrNull() ?: throw PluginError.InvalidCalendarId
        } ?: throw PluginError.CalendarIdMissing
}
